crates/indexer/src/indexer/fetcher.rs

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::counters::{FETCHED_TRANSACTION, UNABLE_TO_FETCH_TRANSACTION};
+use aptos_api::Context;
+use aptos_api_types::{AsConverter, LedgerInfo, Transaction};
+use aptos_logger::prelude::*;
+use aptos_vm::data_cache::StorageAdapterOwned;
+use futures::channel::mpsc;
+use futures::{SinkExt, StreamExt};
+use serde_json::Value;
+use std::sync::Arc;
+use std::time::Duration;
+use storage_interface::state_view::DbStateView;
+use tokio::task::JoinHandle;
+
+// Default Values
+const RETRY_TIME_MILLIS: u64 = 300;
+const MAX_RETRY_TIME_MILLIS: u64 = 120000;
+const TRANSACTION_FETCH_BATCH_SIZE: u16 = 500;
+const TRANSACTION_CHANNEL_SIZE: usize = 35;
+
+#[derive(Debug)]
+pub struct Fetcher {
+    pub context: Arc<Context>,
+    options: TransactionFetcherOptions,
+    chain_id: u8,
+    current_version: u64,
+    highest_known_version: u64,
+    transactions_sender: mpsc::Sender<Vec<Transaction>>,
+}
+
+impl Fetcher {
+    pub fn new(
+        context: Arc<Context>,
+        starting_version: u64,
+        options: TransactionFetcherOptions,
+        transactions_sender: mpsc::Sender<Vec<Transaction>>,
+    ) -> Self {
+        Self {
+            context,
+            options,
+            chain_id: 0,
```

Comment:
why not just pass in chain_id when instantiating?

Reply:
it doesn't exist yet